tests/firecrackerdrivertest.c

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (C) 2021, Amazon, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library;  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include <config.h>
+
+#include "testutils.h"
+
+#include "virlog.h"
+
+#ifdef WITH_FC
+
+#include "vircommand.h"
+
+# define VIR_FROM_THIS VIR_FROM_NONE
+
+VIR_LOG_INIT("tests.firecrackerdrivertest");
+
+static int
+testDriverConnection(const void *opaque G_GNUC_UNUSED)
+{
+    int ret = 0;
+
+    virConnectPtr conn = virConnectOpen("fc:///system");
+
+    if (conn == NULL)
+        ret = -1;
+
+    if (virConnectClose(conn) < 0)
+        ret = -1;
+
+    return ret;
+}
+
+static void
+downloadKernelAndRootFS(void)
+{
+    const char *arch = virArchToString(virArchFromHost());
+    g_autofree const char *kernel_url = g_strdup_printf(
+        "https://s3.amazonaws.com/spec.ccfc.min/img/quickstart_guide/%s/kernels/vmlinux.bin",
+        arch);
+    g_autofree const char *rootfs_url = g_strdup_printf(
+        "https://s3.amazonaws.com/spec.ccfc.min/img/quickstart_guide/%s/rootfs/bionic.rootfs.ext4",
+        arch);
+
+    g_autofree const char *download_kernel_path = g_strdup_printf(
+        "%s/firecrackerdriverdata/hello-vmlinux-test.bin",
+        abs_srcdir);
+    g_autofree const char *download_rootfs_path = g_strdup_printf(
+        "%s/firecrackerdriverdata/hello-rootfs-test.ext4",
+        abs_srcdir);
+
+    virCommand *cmd = virCommandNew("curl");
+
+    virCommandAddArg(cmd, "-fsSL");
+    virCommandAddArg(cmd, "-o");
+    virCommandAddArg(cmd, download_kernel_path);
+    virCommandAddArg(cmd, kernel_url);
+
+    if (virCommandRun(cmd, NULL) < 0) {
+        VIR_WARN("Error downloading test kernel from link: %s",
```

Comment:
you should be returning an error code, so that the test will not continue running if the artifacts weren't downloaded